firstvoices/backend/views/site_feature_views.py

```diff
@@ -0,0 +1,111 @@
+                description=doc_strings.success_200_detail,
+                response=SiteFeatureDetailSerializer,
+            ),
+            403: OpenApiResponse(description=doc_strings.error_403),
+            404: OpenApiResponse(description=doc_strings.error_404),
+        },
+        parameters=[
+            site_slug_parameter,
+            key_parameter,
+        ],
+    ),
+    create=extend_schema(
+        description="Create a new feature flag for the site. Site feature keys cannot be changed after creation.",
+        responses={
+            201: OpenApiResponse(
+                description=doc_strings.success_201,
+                response=SiteFeatureDetailSerializer,
+            ),
+            400: OpenApiResponse(description=doc_strings.error_400_validation),
+            403: OpenApiResponse(description=doc_strings.error_403),
+            404: OpenApiResponse(description=doc_strings.error_404_missing_site),
+        },
+        parameters=[site_slug_parameter],
+    ),
+    update=extend_schema(
+        description="Edit a feature flag for the specified site. Site feature keys cannot be changed after creation.",
+        responses={
+            200: OpenApiResponse(
+                description=doc_strings.success_200_edit,
+                response=SiteFeatureDetailSerializer,
+            ),
+            400: OpenApiResponse(description=doc_strings.error_400_validation),
+            403: OpenApiResponse(description=doc_strings.error_403),
+            404: OpenApiResponse(description=doc_strings.error_404),
+        },
+        parameters=[
+            site_slug_parameter,
+            key_parameter,
+        ],
+    ),
+    partial_update=extend_schema(
+        description="Edit a feature flag for the specified site. Any omitted fields will be unchanged.",
+        responses={
+            200: OpenApiResponse(
+                description=doc_strings.success_200_edit,
+                response=SiteFeatureDetailSerializer,
+            ),
+            400: OpenApiResponse(description=doc_strings.error_400_validation),
+            403: OpenApiResponse(description=doc_strings.error_403),
+            404: OpenApiResponse(description=doc_strings.error_404),
+        },
+        parameters=[
+            site_slug_parameter,
+            key_parameter,
+        ],
+    ),
+    destroy=extend_schema(
+        description="Delete a feature flag from the specified site.",
+        responses={
+            204: OpenApiResponse(description=doc_strings.success_204_deleted),
+            403: OpenApiResponse(description=doc_strings.error_403),
+            404: OpenApiResponse(description=doc_strings.error_404),
+        },
+        parameters=[site_slug_parameter, key_parameter],
+    ),
+)
+class SiteFeatureViewSet(
+    SiteContentViewSetMixin, FVPermissionViewSetMixin, ModelViewSet
+):
+    """
+    API endpoint that allows site feature flags to be viewed or edited.
+    """
+
+    lookup_field = "key"
```

Comment:
If the lookup field is the key and the key is allowed to have any characters then the detail view will have the strange characters in the slug (eg: if the key is "test key" then the slug will be `.../<site-slug>/features/test key`, which we probably don't want).

We may want to either restrict the format of the key field or add another field to hold the slug.